src/client/components/header/search/Search.js

```diff
@@ -66,4 +85,19 @@ class Search extends React.Component {
   }
 }
 
+let mapStateToProps = (state) => {
+  return {
+    search: state.searchState,
+  }
+}
+
+let mapDispatchToProps = (dispatch) => {
+  return {
+    getSearchMoviesByTitle: (searchText) => { dispatch(getSearchMoviesByTitleAction(searchText)) },
+    getSearchMoviesByGengre: (searchText) => { dispatch(getSearchMoviesByGengreAction(searchText)) },
+  }
+}
+
+let Search = connect(mapStateToProps, mapDispatchToProps)(SearchAPI);
```

Comment:
I propose you create high order component and pass all this store variables into Search component